### ISSUE TITLE
Can't splice in AnyVal type as a parent type

### ISSUE THREAD

**Issue by dwijnand** [scalameta/scalameta#448]
Here's an example:

```scala
package p

import scala.{ collection => sc }, sc.{ immutable => sci }, sci.{ Seq => sciSeq }

import scala.meta._

class bob extends scala.annotation.StaticAnnotation {
  inline def apply(defn: Any) = meta {
    val q"""
      ..$mods0 class $tname[..$tparams] ..$ctorMods (...$paramss0) extends { ..$earlyStats } with ..$ctorcalls {
        $selfParam =>
        ..$stats
      }
    """ = defn

    val name = Term.Name(tname.value)

    val AnyVal = t"_root_.scala.AnyVal"

    val objectDefn = q"""
      object $name {
        final class Builder(val x: Int) extends $AnyVal
      }
    """

    Term.Block(sciSeq(defn.asInstanceOf[Stat], objectDefn))
  }
}
```

You can simulate the bug by running this:

```scala
package p

@bob class Foo(x: Int)

object Test {
  def main(args: Array[String]): Unit = {
    println(new Foo.Builder(1))
  }
}
```

**Comment by dwijnand** [comment]
Using "extends _root_.scala.AnyVal" instead works.

The error is:

```scala
[error] /s/scalameta-anyval-bug/src/main/scala/p/bob.scala:22: type mismatch when unquoting;
[error]  found   : scala.meta.Type.Select
[error]  required: scala.meta.Template
[error]         final class Builder(val x: Int) extends $AnyVal
[error]                                                 ^
[error] one error found
```

**Comment by xeno-by** [comment]
I've pushed a workaround described in details in the commit that got linked to this issue. It'll be available under 1.1.0-SNAPSHOT within 15-30 minutes.
